lighty-examples/lighty-gnmi-community-restconf-app/src/test/java/io/lighty/examples/controllers/gnmi/GnmiRestconfExampleTest.java

```diff
@@ -0,0 +1,312 @@
+    private static final String KEYSTORE_ID = "keystore-id-1";
+    private static final String USERNAME = "Admin";
+    private static final String PASSWORD = "Admin";
+    private static final String SYSTEM_AUTHENTICATION = "openconfig-system:authentication";
+    private static final String STATE = "state";
+    private static final String CONFIG = "config";
+    private static final String AUTHENTICATION_METHOD = "authentication-method";
+
+    private static final String RESTCONF_PATH = "http://localhost:8888/restconf";
+    private static final String ADD_CERTIFICATE_PATH
+            = RESTCONF_PATH + "/operations/gnmi-certificate-storage:add-keystore-certificate";
+    private static final String MOUNTPOINT_PATH
+            = RESTCONF_PATH + "/data/network-topology:network-topology/topology=gnmi-topology/node=" + DEVICE_ID;
+    private static final String MOUNTPOINT_STATUS_PATH = MOUNTPOINT_PATH + "/gnmi-topology:node-state/node-status";
+    private static final String NODE_STATUS_RESPONSE_READY = "{\"gnmi-topology:node-status\":\"READY\"}";
+    private static final String GET_INTERFACES = MOUNTPOINT_PATH + "/yang-ext:mount/openconfig-interfaces:interfaces";
+    private static final String GET_SYSTEM = MOUNTPOINT_PATH + "/yang-ext:mount/openconfig-system:system";
+    private static final String GET_AUTHENTICATION
+            = MOUNTPOINT_PATH + "/yang-ext:mount/openconfig-system:system/aaa/authentication";
+    private static final String GET_AUTHENTICATION_CONFIG = GET_AUTHENTICATION + "/config" ;
+    private static final String GET_CONFIG_AUTHENTICATION_CONFIG = GET_AUTHENTICATION_CONFIG + "?content=config" ;
+    private static final String GET_CONFIG_AUTHENTICATION = GET_AUTHENTICATION + "?content=config";
+
+    private static ExecutorService httpClientExecutor;
+    private static HttpClient httpClient;
+
+    @BeforeAll
+    public static void startUp() throws IOException {
+        httpClientExecutor = Executors.newSingleThreadExecutor();
+        httpClient = HttpClient.newBuilder().executor(httpClientExecutor).build();
+
+        Main.main(new String[]{});
+    }
+
+    @AfterAll
+    public static void teardown() throws IOException, InterruptedException {
+        sendDeleteRequest(MOUNTPOINT_PATH);
+        httpClientExecutor.shutdownNow();
+    }
+
+    @Test
+    public void readmeExampleTest() throws IOException, InterruptedException {
+        // Register keystore.
+        final String certificatesRequestBody = getCertificatesRequestBody(getResources(CA_CERTIFICATE_PATH),
+                getResources(CLIENT_KEY_PATH), getResources(CLIENT_CERTIFICATE_PATH));
+        final HttpResponse<String> response = sendPostRequest(ADD_CERTIFICATE_PATH, certificatesRequestBody);
+        assertEquals(response.statusCode(), HttpURLConnection.HTTP_NO_CONTENT);
+
+        // Check if device was not created before.
+        HttpResponse<String> notCreatedDevice = sendGetRequest(MOUNTPOINT_STATUS_PATH);
```

Comment:
for responses maybe add *Response to be more readable

Reply:
done